### ISSUE TITLE
src directory not idiomatic Go?

### ISSUE THREAD

**Issue by domdom82** [notsatan/go-template#36]
Hi,

first of all, I really like your project. I've been searching for a ready-to-use template for go projects for some time and so far have only found [cookiecutter-golang](https://github.com/lacion/cookiecutter-golang) but I like yours more.

However, I am a bit puzzled by the generated `src` folder inside the generated project. That seems to me non-idiomatic go as described [here](https://github.com/golang-standards/project-layout#src) 

In my case I would probably delete the `src` folder after the project has been created.

Can you explain your reasoning behind using such a folder? I'm just trying to understand why you think it's needed.

**Comment by notsatan** [comment]
> However, I am a bit puzzled by the generated `src` folder inside the generated project
> [...]
> Can you explain your reasoning behind using such a folder?

**Tl;dr:** To avoid cluttering code files with non-code files.

While generating a project using `go-template`, if you choose to enable all features - you'll end up with a bunch of non-code files in the home directory - for example, take a look at the [initial commit][init-commit] for [`notsatan/crcgen`][crcgen] - one of my side projects generated with *go-template*.

Trying to look for `config.go` between `CODE_OF_CONDUCT.md` and `CONTRIBUTING.md` just becomes annoying at some point in time. As such, when you start out with a bunch of non-code files and then add your code files at the same place, you'll just end up cluttering them together.

> That seems to me non-idiomatic go as described here

True. But the guide itself [isn't "official"][issue] by any means, as such I won't be using it to define idiomatic Go.

It's very annoying that the repo is named `golang-standards/project-layout` (which just leads to people following the guide religiously, only to end up in a mess).

> I'm just trying to understand why you think it's needed

Developers aiming to create a basic project in Go will stick to `go mod init`. 

On the other hand, for larger projects (>10K LOC), people usually have a detailed idea of a project structure to start out with.

The target audience for `go-template` would be using it to generate medium-sized projects with multiple modules and files, all of which can be comfortably nested within the `src` directory - something I usually do for most of my projects. In my experience, having a single "root" directory containing all the code files is really better than multiple code directories, or dumping code files in the base directory - of course, this might not work for large-scale projects.

[init-commit]: https://github.com/notsatan/crcgen/tree/e472e7f3f0bfe21b5f66334df58817c7b9d9255a
[crcgen]: https://github.com/notsatan/crcgen
[issue]: https://github.com/golang-standards/project-layout/issues/117

**Comment by notsatan** [comment]
> Which just leads to people following the guide religiously, only to end up in a mess

To expand on this, I wouldn't recommend anyone to strictly follow all the points mentioned in `golang-standards/project-layout` - more so, I've frequently found myself disagreeing with many of the points they mention. 

This, combined with the repo trying to push itself as "golang-standards", is why I decided to skip mentioning it in the [resources section][res] of `go-template`.

As a fun fact, [`golang/go`][go-repo] - the source code for Go programming language - has an `src` directory containing most (if not all) of the source code for Go :p

[go-repo]: https://github.com/golang/go
[res]: https://github.com/notsatan/go-template#dart-whats-next

**Comment by domdom82** [comment]
Thanks!
